sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -254,9 +258,10 @@ object SharedState extends Logging {
       // the value of spark.sql.warehouse.dir.
       // When neither spark.sql.warehouse.dir nor hive.metastore.warehouse.dir is set
       // we will set hive.metastore.warehouse.dir to the default value of spark.sql.warehouse.dir.
-      val sparkWarehouseDir = sparkConf.get(WAREHOUSE_PATH)
+      val sparkWarehouseDir = sparkWarehouse.getOrElse(sparkConf.get(WAREHOUSE_PATH))
```

Comment:
`sparkWarehouse` already has `.orElse(sparkConf.getOption(WAREHOUSE_PATH.key))`. Do you mean `sparkWarehouse.getOrElse(WAREHOUSE_PATH.defaultValueString)`?

Reply:
Yes, here will  finally touch the default value string if never matched before

Reply:
can we change to `sparkWarehouse.getOrElse(WAREHOUSE_PATH.defaultValueString)` so that it's more explicit?

Reply:
SGTM